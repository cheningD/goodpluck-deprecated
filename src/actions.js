import { updateSignedInUserInLocalStorage } from "./util"

export const createUser = async params => {
  console.log("Sending data to backend... ", params)
  const response = await fetch("https://api.goodpluck.com/createuser", {
    method: "POST",
    headers: {
      "Content-Type": "application/json",
    },
    body: JSON.stringify(params),
  })
  const responseJson = await response.json()

  return responseJson
}

export const signIn = async params => {
  console.log("Sending data to backend... ", params)
  const response = await fetch("https://api.goodpluck.com/signin", {
    method: "POST",
    headers: {
      "Content-Type": "application/json",
    },
    body: JSON.stringify(params),
  })
  const responseJson = await response.json()

  return responseJson
}

export const checkEmailVerificationAndSignIn = async (nonce, email) => {
  const response = await fetch("https://api.goodpluck.com/checkemailverify", {
    method: "POST",
    headers: {
      "Content-Type": "application/json",
    },
    body: JSON.stringify({ none: nonce, email: email }),
  })
  if (response.status !== 200) {
    console.log("Something went wrong checking email")
    return false
  } else {
    const responseJson = await response.json()
    if (responseJson.signedInUser) {
      updateSignedInUserInLocalStorage(responseJson.signedInUser)
      return true
    } else {
      console.log("You havent verified your email yet.", responseJson)
      return false
    }
  }
}
