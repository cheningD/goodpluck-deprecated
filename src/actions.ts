import {
  BasketItemData,
  CheckEmailAndZipJSONResponse,
  CreateUserSuccessResponseJSON,
  GoodpluckJSONResponse,
  OrderDetail,
  SignedInData,
  StripeCustomer,
} from './types'

import { navigate } from 'gatsby'
import toast from 'react-hot-toast'

// For testing the api on localhost
export const LOCAL_API_PREFIX =
  process.env.GATSBY_DEPLOY_ENVIRONMENT === 'DEVELOPMENT' ? 'https://localhost.goodpluck.com' : ''

export const createUser = async (params: Record<string, any>): Promise<CreateUserSuccessResponseJSON> => {
  const response = await fetch(`${LOCAL_API_PREFIX}/api/createuser`, {
    credentials: 'same-origin',
    method: 'POST',
    headers: {
      'Content-Type': 'application/json',
    },
    body: JSON.stringify(params),
  })

  const responseJson = await response.json()

  return responseJson
}

export const signIn = async params => {
  const response = await fetch(`${LOCAL_API_PREFIX}/api/signin`, {
    credentials: 'same-origin',
    method: 'POST',
    headers: {
      'Content-Type': 'application/json',
    },
    body: JSON.stringify(params),
  })
  return response
}

export const signInAsTestUser = async (): Promise<any> => {
  const response = await fetch(`${LOCAL_API_PREFIX}/api/signinastestuser`, {
    credentials: 'same-origin',
    method: 'POST',
    headers: {
      'Content-Type': 'application/json',
    },
    body: JSON.stringify({}),
  })

  let data

  try {
    data = await response.json()
  } catch (err) {
    return response
  }

  return data
}

/** If parameters validate, client recieves a session cookie
 *
 * @param authCodeId - The ID of the AuthCode created at sign in
 * @param code - The 'secret' part of the AuthCode
 * @param email - The email logging in
 * @returns {boolean} - true iof you are now signed in
 */
export const verifyEmail = async (
  authCodeId: string,
  code: string,
  email: string,
): Promise<GoodpluckJSONResponse | null> => {
  const response = await fetch(`${LOCAL_API_PREFIX}/api/verifyemail`, {
    credentials: 'same-origin',
    method: 'POST',
    headers: {
      'Content-Type': 'application/json',
    },
    body: JSON.stringify({ authCodeId, code, email: email.toLowerCase() }),
  })

  try {
    const responseJSON = await response.json()
    return responseJSON
  } catch (err) {
    console.log(`Error in verify email: ${err.message || err}`)
    return { error: 'Something went wrong, please try again' }
  }
}

export const checkEmailZip = async (email: string, zip: string): Promise<CheckEmailAndZipJSONResponse> => {
  const response = await fetch(`${LOCAL_API_PREFIX}/api/checkemail`, {
    credentials: 'same-origin',
    method: 'POST',
    headers: {
      'Content-Type': 'application/json',
    },
    body: JSON.stringify({ email: email.toLowerCase(), zip }),
  })
  return await response.json()
}

export const getSignedInData = async (): Promise<SignedInData | null> => {
  const response = await fetch(`${LOCAL_API_PREFIX}/api/getsignedinuser`, {
    credentials: 'same-origin',
  })

  if (!response.ok) {
    return null
  }

  try {
    const responseJSON = await response.json()
    return responseJSON.data
  } catch (err) {
    console.log(`Error in getSignedInData: ${err.message || err}`)
    return null
  }
}

export const retrieveCustomer = async (): Promise<StripeCustomer | null> => {
  const response = await fetch(`${LOCAL_API_PREFIX}/api/retrievecustomer`, {
    credentials: 'same-origin',
  })

  if (!response.ok) {
    return null
  }

  try {
    const responseJSON = await response.json()
    return responseJSON.data
  } catch (err) {
    console.log(`Error in retrieveCustomer: ${err.message || err}`)
    return null
  }
}

export const restartSubscription = async (): Promise<GoodpluckJSONResponse> => {
  const response = await fetch(`${LOCAL_API_PREFIX}/api/restartsubscription`, {
    credentials: 'same-origin',
    method: 'POST',
    headers: {
      'Content-Type': 'application/json',
    },
  })

  try {
    return await response.json()
  } catch (err) {
    console.log(`Error in restartSubscription: ${err.message || err}`)
    return {
      success: false,
      error: 'Something went wrong. Please contact us if this issue persists',
    }
  }
}

export const pauseSubscription = async (reason: string): Promise<GoodpluckJSONResponse> => {
  const response = await fetch(`${LOCAL_API_PREFIX}/api/pausesubscription`, {
    credentials: 'same-origin',
    method: 'POST',
    headers: {
      'Content-Type': 'application/json',
    },
    body: JSON.stringify({ reason: reason }),
  })

  try {
    return await response.json()
  } catch (err) {
    console.log(`Error in pauseSubscription: ${err.message || err}`)
    return {
      success: false,
      error: 'Something went wrong. Please contact us if this issue persists',
    }
  }
}

export const getSetSkippedFunc = (orders, setOrders: Function) => {
  const setSkipped = async (orderMondayIndex: string, skip: boolean) => {
    const response = await editOrder(orderMondayIndex, skip)
    let data: Record<string, any>
    try {
      data = (await response.json()).data
    } catch (err) {}

    if (data && data.updatedOrder) {
      setOrders(Object.assign({}, orders, { [orderMondayIndex]: data.updatedOrder }))
    } else if (data && data.error) {
      toast.error(`That didn't work: ${data.error || ''}`)
    } else {
      toast.error('That didnt work: Something went wrong when we tried to edit this order.')
      // Todo: Alert sentry
    }
  }
  return setSkipped
}

export const editOrder = async (orderMondayIndex: string, skip: boolean): Promise<Response> => {
  return await fetch(`${LOCAL_API_PREFIX}/api/editorder`, {
    credentials: 'same-origin',
    method: 'POST',
    headers: {
      'Content-Type': 'application/json',
    },
    body: JSON.stringify({ orderMondayIndex, skip }),
  })
}

export const updateStripeCard = async (tokenID: string): Promise<Record<string, string> | null> => {
  const response = await fetch(`${LOCAL_API_PREFIX}/api/updatecard`, {
    credentials: 'same-origin',
    method: 'POST',
    headers: {
      'Content-Type': 'application/json',
    },
    body: JSON.stringify({ tokenID: tokenID }),
  })

  if (!response.ok) {
    console.log(`Error in updateStripeCard: ${response.statusText}`)
    return null
  }

  try {
    const responseJSON: GoodpluckJSONResponse = await response.json()
    return responseJSON.data
  } catch (err) {
    console.log(`Error in updateStripeCard: ${err.message || err}`)
    return null
  }
}

export const getOrders = async (): Promise<Record<string, OrderDetail> | null> => {
  const response = await fetch(`${LOCAL_API_PREFIX}/api/orders`, {
    credentials: 'same-origin',
  })

  if (!response.ok) {
    return null
  }

  try {
    const responseJSON: GoodpluckJSONResponse = await response.json()
    return responseJSON.data
  } catch (err) {
    console.log(`Error in getOrders: ${err.message || err}`)
    return null
  }
}

export const getOldOrders = async (): Promise<Record<string, OrderDetail> | null> => {
  const response = await fetch(`${LOCAL_API_PREFIX}/api/pastorders`, {
    credentials: 'same-origin',
  })

  if (!response.ok) {
    return null
  }

  try {
    const responseJSON: GoodpluckJSONResponse = await response.json()
    return responseJSON.data
  } catch (err) {
    console.log(`Error in getOrders: ${err.message || err}`)
    return null
  }
}

export const getOrdersDemo = async (): Promise<OrderDetail | null> => {
  const response = await fetch(`${LOCAL_API_PREFIX}/api/orders/demo`, {
    credentials: 'same-origin',
  })

  if (!response.ok) {
    return null
  }

  try {
    const responseJSON: GoodpluckJSONResponse = await response.json()
    return responseJSON.data as OrderDetail
  } catch (err) {
    console.log(`Error in getOrdersDemo: ${err.message || err}`)
    return null
  }
}

export const getBasket = async (): Promise<Map<string, BasketItemData> | null> => {
  const response = await fetch(`${LOCAL_API_PREFIX}/api/basket`, {
    credentials: 'same-origin',
  })

  if (!response.ok) {
    console.log(`Error in getBasket: ${response}`)
    return null
  }

  try {
    const responseJSON = await response.json()
    const newBasket: Map<string, BasketItemData> = new Map(JSON.parse(responseJSON.data))
    return newBasket
  } catch (err) {
    console.log(`Error in getBasket: ${err.message || err}`)
    return null
  }
}

export const updateBasket = async (basket: Map<string, BasketItemData>): Promise<boolean> => {
  const response = await fetch(`${LOCAL_API_PREFIX}/api/basket`, {
    credentials: 'same-origin',
    method: 'POST',
    headers: {
      'Content-Type': 'application/json',
    },
    body: JSON.stringify({ data: JSON.stringify(Array.from(basket.entries())) }),
  })

  try {
    const responseJSON = await response.json()
    if (responseJSON.data) {
      return true
    }
  } catch (err) {
    console.log(`Error in updateBasket: ${err.message || err}`)
    return false
  }
  return false
}

export const logout = async () => {
  if (typeof localStorage !== `undefined`) {
    localStorage.clear()
  }

  await fetch(`${LOCAL_API_PREFIX}/api/logout`, {
    credentials: 'same-origin',
  })
  navigate('/')
}
