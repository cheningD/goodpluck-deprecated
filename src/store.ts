import { BasketItemData, OrderSupabase, StripeCustomer, SubscriptionRecord } from './types'
import { atom, selector } from 'recoil'

export const signedInUser = atom({
  key: 'signed_in_user', // unique ID (with respect to other atoms/selectors)
  default: null, // default value (aka initial value),
})

const stripeCustomerDefault: StripeCustomer | null = null
export const stripeCustomer = atom({
  key: 'stripeCustomer', // unique ID (with respect to other atoms/selectors)
  default: stripeCustomerDefault, // default value (aka initial value),
})

export const isSignedIn = selector({
  key: 'isSignedIn',
  get: ({ get }) => {
    const user = get(signedInUser)
    if (user !== null && user.email) {
      return true
    }
    return false
  },
})

export const myOrders = atom({
  key: 'orders', // unique ID (with respect to other atoms/selectors)
  default: null, // default value (aka initial value),
})

export const pastOrders = atom({
  key: 'pastorders', // unique ID (with respect to other atoms/selectors)
  default: null, // default value (aka initial value),
})

const ordersDefault: OrderSupabase[] | null = null

export const pastOrders2 = atom({
  key: 'pastorders2', // unique ID (with respect to other atoms/selectors)
  default: ordersDefault, // default value (aka initial value),
})

const subscriptionsDefault: SubscriptionRecord[] | null = null

export const mySubscriptions = atom({
  key: 'subscriptions', // unique ID (with respect to other atoms/selectors)
  default: subscriptionsDefault, // default value (aka initial value)
})

const basketDefault: Map<string, BasketItemData> | null = null

export const basketItems = atom({
  key: 'basket_items', // unique ID (with respect to other atoms/selectors)
  default: basketDefault, // default value (aka initial value)
})

export const basketCount = selector({
  key: 'basketCount',
  get: ({ get }) => {
    if (get(basketItems) === null) {
      return 0
    }
    return get(basketItems).size
  },
})

export const subtotalInCents = selector({
  key: 'subtotalInCents',
  get: ({ get }) => {
    let priceInCents: number = 0
    if (get(basketItems) === null) {
      return 0
    }
    get(basketItems).forEach(item => {
      priceInCents = priceInCents + item.quantity * item.unitPriceInCents
    })
    return priceInCents
  },
})

export const shippingInCents = selector({
  key: 'shippingInCents',
  get: ({ get }) => {
    if (get(subtotalInCents) === 0 || get(subtotalInCents) >= 3000) {
      return 0
    } else {
      return 699 // $6.99
    }
  },
})

// Onboarding Quiz Data
export const onboardingEmail = atom({
  key: 'onboardingEmail',
  default: '', // Empty string is correct default for text forms
})

export const onboardingZip = atom({
  key: 'onboardingZip', // unique ID (with respect to other atoms/selectors)
  default: '', // Empty string is correct default for text forms
})

export const onboardingShoppingFor = atom({
  key: 'onboardingShoppingFor', // unique ID (with respect to other atoms/selectors)
  default: [], // Empty array of strings is correct default for checkbox group
})

export const onboardingMyCauses = atom({
  key: 'onboardingMyCauses', // unique ID (with respect to other atoms/selectors)
  default: [], // Empty array of strings is correct default for checkbox group
})

export const onboardingOrderFrequency = atom({
  key: 'onboardingOrderFrequency',
  default: 'every week',
})

export const onboardingNotifyDeliveryDates = atom({
  key: 'onboardingNotifyDeliveryDates',
  default: false,
})
