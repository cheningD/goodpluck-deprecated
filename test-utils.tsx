import React from 'react'
import { render, RenderOptions } from '@testing-library/react'
import { RecoilRoot } from 'recoil'

type Options = Omit<RenderOptions, 'queries'>

type InitializeState = Array<{
  atom: any
  value: any
}>

const customRender = (ui: React.ReactElement<any>, options?: Options, state?: InitializeState) =>
  render(ui, {
    wrapper: ({ children }) => {
      return (
        <RecoilRoot
          initializeState={({ set }) => {
            if (state?.length) {
              state.forEach(atomState => set(atomState.atom, atomState.value))
            }
          }}
        >
          {children}
        </RecoilRoot>
      )
    },
    ...options,
  })

// re-export everything
export * from '@testing-library/react'

// override render method
export { customRender as render }
