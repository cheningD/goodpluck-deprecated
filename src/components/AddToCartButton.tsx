import { BasketItemData, OrderDetail } from '../types'
import { basketItems, myOrders } from '../store'
import { useRecoilState, useRecoilValue } from 'recoil'

import { DateTime } from 'luxon'
import React from 'react'
import Select from 'react-select'
import styled from 'styled-components'
import { updateBasket } from '../actions'

const Submit = styled.button`
  background-color: #fff;
  border-radius: 4px;
  border: 1px solid hsl(0, 0%, 80%);
  color: #333;
  display: inline-block;
  font-family: hk_grotesksemibold, sans-serif;
  font-size: 0.875rem;
  height: 38px;
  margin-right: 16px;
  width: 7rem;

  &:hover,
  &:focus {
    background-color: #788474;
    color: #fff;
  }
`

const AddToCartSelect = styled(Select)`
  display: inline-block;
  font-family: hk_grotesksemibold, sans-serif;
  font-size: 0.875rem;
  margin-right: 16px;
  width: 7rem;
`

const AddToCartButton = ({ stripePriceId, unitPriceInCents, quantityInBasket }) => {
  const orders = useRecoilValue(myOrders)

  let upcomingOrderData: OrderDetail | null = null
  if (orders && Object.keys(orders).length > 0) {
    upcomingOrderData = orders[Object.keys(orders).slice().sort()[0]]
  }

  let canEditBasket: boolean = true
  let basketConfirmationMessage = ''
  const dateNow = DateTime.local()

  // If editBasketEndDate is in the past, dont let them edit....
  if (upcomingOrderData && DateTime.fromISO(upcomingOrderData.editBasketEndDate) < DateTime.local()) {
    return <div>{`Quantity: ${quantityInBasket}`}</div>
  }

  const [basket, _setBasket] = useRecoilState(basketItems)
  const setBasket = async (basket: Map<string, BasketItemData>) => {
    updateBasket(basket)
    _setBasket(basket)
  }
  if (quantityInBasket === 0) {
    return (
      <Submit
        as="button"
        onClick={() => {
          let newBasket: Map<string, BasketItemData> = new Map(basket)
          newBasket.set(stripePriceId, { stripePriceId, quantity: 1, unitPriceInCents })
          setBasket(newBasket)
        }}
      >
        Add
      </Submit>
    )
  }

  const options = [
    { value: 0, label: '0 (Remove)' },
    { value: 1, label: '1' },
    { value: 2, label: '2' },
    { value: 3, label: '3' },
    { value: 4, label: '4' },
    { value: 5, label: '5' },
    { value: 6, label: '6' },
    { value: 7, label: '7' },
    { value: 8, label: '8' },
    { value: 9, label: '9' },
  ]

  return (
    <AddToCartSelect
      classNamePrefix="add-to-cart-select"
      options={options}
      onChange={option => {
        if (option.value === 0) {
          let newBasket: Map<string, BasketItemData> = new Map(basket)
          newBasket.delete(stripePriceId)
          setBasket(newBasket)
        } else {
          let newBasket: Map<string, BasketItemData> = new Map(basket)
          newBasket.set(stripePriceId, { stripePriceId, quantity: option.value, unitPriceInCents })
          setBasket(newBasket)
        }
      }}
      isSearchable={false}
      components={{
        IndicatorSeparator: null,
      }}
      value={{
        label: `Qty: ${quantityInBasket}`,
        value: quantityInBasket,
      }}
    />
  )
}

export default AddToCartButton
