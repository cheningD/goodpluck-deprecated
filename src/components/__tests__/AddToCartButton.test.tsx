import React from 'react'
import selectEvent from 'react-select-event'
import { rest } from 'msw'
import { setupServer } from 'msw/node'

import AddToCartButton from '../AddToCartButton'
import { render } from '../../../test-utils'
import * as actions from '../../actions'
import { LOCAL_API_PREFIX } from '../../actions'
import { basketItems } from '../../store'

const STRIPE_PRICE_ID = '1'

const defaultProps = {
  stripePriceId: STRIPE_PRICE_ID,
  unitPriceInCents: '20',
  quantityInBasket: 1,
}

/* API calls */

const server = setupServer(
  rest.post(`${LOCAL_API_PREFIX}/api/basket`, (req, res, ctx) => {
    return res(
      ctx.json({
        data: [
          {
            stripePriceId: '1',
            quantity: 5,
            unitPriceInCents: 5,
          },
        ],
      }),
    )
  }),
)

describe('AddToCartButton', () => {
  beforeAll(() => server.listen())
  afterEach(() => {
    server.resetHandlers()
    jest.clearAllMocks()
  })
  afterAll(() => server.close())

  it('renders correctly', () => {
    const { container } = render(<AddToCartButton {...defaultProps} />)
    expect(container.firstChild).toMatchSnapshot()
  })

  it('renders add button when quantityInBasket is 0', () => {
    const props = {
      ...defaultProps,
      quantityInBasket: 0,
    }
    const { getByText } = render(<AddToCartButton {...props} />)
    expect(getByText('Add')).toBeTruthy()
  })

  it('renders the select with options when quantityInBasket is greater than 0', () => {
    const props = {
      ...defaultProps,
      quantityInBasket: 1,
    }
    const { getByLabelText } = render(<AddToCartButton {...props} />)
    expect(getByLabelText('add to cart')).toBeTruthy()
  })

  it('removes the item from basket when selecting option with value 0', async () => {
    const mockedUpdateBasket = jest.spyOn(actions, 'updateBasket')
    const props = {
      ...defaultProps,
      quantityInBasket: 1,
    }
    const basketItem = new Map()
    basketItem.set(STRIPE_PRICE_ID, {
      stripePriceId: STRIPE_PRICE_ID,
      quantity: 1,
      unitPriceInCents: 10,
    })
    const { container, getByLabelText } = render(<AddToCartButton {...props} />, null, [
      {
        atom: basketItems,
        value: basketItem,
      },
    ])
    await selectEvent.select(getByLabelText('add to cart'), ['0 (Remove)'])
    const params = mockedUpdateBasket.mock.calls[0][0]
    expect(mockedUpdateBasket).toHaveBeenCalledTimes(1)
    expect(params.size).toEqual(0)
  })
  it('updates the item from basket when selecting option with value greater than 0', async () => {
    const mockedUpdateBasket = jest.spyOn(actions, 'updateBasket')
    const props = {
      ...defaultProps,
      quantityInBasket: 5,
    }
    const basketItem = new Map()
    basketItem.set(STRIPE_PRICE_ID, {
      stripePriceId: STRIPE_PRICE_ID,
      quantity: 1,
      unitPriceInCents: 10,
    })
    const { container, getByLabelText } = render(<AddToCartButton {...props} />, null, [
      {
        atom: basketItems,
        value: basketItem,
      },
    ])
    await selectEvent.select(getByLabelText('add to cart'), ['5'])
    const params = mockedUpdateBasket.mock.calls[0][0]
    expect(mockedUpdateBasket).toHaveBeenCalledTimes(1)
    expect(params.get(STRIPE_PRICE_ID).quantity).toEqual(5)
  })
})
