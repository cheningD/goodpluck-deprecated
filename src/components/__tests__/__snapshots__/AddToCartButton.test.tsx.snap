// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddToCartButton renders correctly 1`] = `
<div
  class="sc-gsTCUz gYIOIB css-2b097c-container"
>
  <div
    class="add-to-cart-select__control css-yk16xz-control"
  >
    <div
      class="add-to-cart-select__value-container add-to-cart-select__value-container--has-value css-g1d714-ValueContainer"
    >
      <div
        class="add-to-cart-select__single-value css-1uccc91-singleValue"
      >
        Qty: 1
      </div>
      <input
        aria-autocomplete="list"
        aria-label="add to cart"
        class="css-62g3xt-dummyInput"
        id="react-select-2-input"
        readonly=""
        tabindex="0"
        value=""
      />
    </div>
    <div
      class="add-to-cart-select__indicators css-1hb7zxy-IndicatorsContainer"
    >
      <div
        aria-hidden="true"
        class="add-to-cart-select__indicator add-to-cart-select__dropdown-indicator css-tlfecz-indicatorContainer"
      >
        <svg
          aria-hidden="true"
          class="css-6q0nyr-Svg"
          focusable="false"
          height="20"
          viewBox="0 0 20 20"
          width="20"
        >
          <path
            d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
          />
        </svg>
      </div>
    </div>
  </div>
</div>
`;
